e2e/util/crud.go

```diff
@@ -49,9 +50,26 @@ func DeleteNamespace(client client.Client, namespace string) error {
 		}
 
 		Ctx.Log.Info("namespace " + namespace + " not found")
+
+		return nil
 	}
 
-	return nil
+	Ctx.Log.Info("waiting until namespace " + namespace + " is deleted")
+	key := types.NamespacedName{Name: namespace}
+
+	for {
```

Comment:
So this loop runs forever. No timeout, and no indication that it is running forever in case it is stuck deleting the namespace.

Reply:
There is an indication:

```go
"waiting until namespace " + namespace + " is deleted")
```

We can add a timeout, maybe using our standard timeout?

Reply:
BTW in the CI this job will fail with a timeout from our retry action. In a local run, you will have to interrupt the test process. Even if we add a timeout, you will still have to do manual cleanup to fix the env.

Reply:
Sure, but from the code, it appears that deleting the namespace is the final step. This means the namespace is empty and should take no more than 5 seconds to delete. However, if the CI job times out and this message is kept as the last log entry, that should be fine.

Reply:
Turns out we don't use timeout for e2e tests, but this is easy to add and good idea anyway. But I think adding a timeout makes sense anyway and is simple.

Reply:
Added timeout, ci timeout and retries added in #1626